File: src/main/java/com/uwetrottmann/trakt5/entities/MovieIds.java
package com.uwetrottmann.trakt5.entities;

import javax.annotation.Nonnull;

public class MovieIds extends BaseIds {

    public String slug;

    @Nonnull
    public static MovieIds trakt(int trakt) {
        MovieIds ids = new MovieIds();
        ids.trakt = trakt;
        return ids;
    }

    @Nonnull
    public static MovieIds imdb(String imdb) {
        MovieIds ids = new MovieIds();
        ids.imdb = imdb;
        return ids;
    }

    @Nonnull
    public static MovieIds tmdb(int tmdb) {
        MovieIds ids = new MovieIds();
        ids.tmdb = tmdb;
        return ids;
    }

    @Nonnull
    public static MovieIds slug(String slug) {
        MovieIds ids = new MovieIds();
        ids.slug = slug;
        return ids;
    }

}

File: src/main/kotlin/com/redissi/trakt/entities/MovieIds.kt
package com.redissi.trakt.entities

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
class MovieIds(
    override val trakt: Int? = null,
    override val imdb: String? = null,
    override val tmdb: Int? = null,
    val slug: String? = null
) : BaseIds() {

    companion object {
        fun trakt(trakt: Int): MovieIds {
            return MovieIds(trakt = trakt)
        }

        fun imdb(imdb: String?): MovieIds {
            return MovieIds(imdb = imdb)
        }

        fun tmdb(tmdb: Int): MovieIds {
            return MovieIds(tmdb = tmdb)
        }

        fun slug(slug: String?): MovieIds {
            return MovieIds(slug = slug)
        }
    }
}